C/c4CAPI.cc

```diff
@@ -147,7 +151,9 @@ size_t c4stream_read(C4ReadStream* stream, void* buffer, size_t maxBytes, C4Erro
 
 int64_t c4stream_getLength(C4ReadStream* stream, C4Error* outError) noexcept {
     try {
-        return stream->getLength();
+        uint64_t streamLength = stream->getLength();
+        DebugAssert(streamLength < std::numeric_limits<int64_t>::max());
+        return static_cast<int64_t>(streamLength);
```

Comment:
I actually already have a facility for this, it is called `narrow_cast` in NumConversion.hh

Reply:
Probably better use narrow_cast, which checks the bounds.

Reply:
Oh awesome, thanks.